### PR TITLE
Unified REST API: domains

### PR DESCRIPTION
**Pull request by chrzaszcz**
Integrate the domain REST handler with `mongoose_admin_api`

The domain handler had some custom code:
- The auth failed if creds were provided, but not configured. I decided to apply this behaviour to the whole Admin API.
- Authorization and request parsing. I just used the functionality already provided by `mongoose_admin_api`.
- `{"what": "message"}` error messages. I decided to use the plain-text ones as in `mongoose_admin_api`.
- Custom test helpers. I also decided to use the ones for `mongoose_admin_api`.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/138296) / small_tests / 05a93cf78e0f2df686559827eb4ca2e8c86b8208
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3768/138296/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3768/138296/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-22_07.15.17/lib.mongooseim.logs/run.2022-09-22_07.15.17/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/138321) / small_tests / 2e3e234306d3a2d60efce6259b6dad1ec2253b38
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3768/138321/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3768/138321/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-22_07.48.59/lib.mongooseim.logs/run.2022-09-22_07.49.00/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/138351) / small_tests / 33b5d9c14bd9800eb3f60da60a9905f2cc638094
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3768/138351/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3768/138351/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-22_08.45.45/lib.mongooseim.logs/run.2022-09-22_08.45.46/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/138398) / small_tests / 9082e6e30ba500766800836ae0f2e83e40f0093f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3768/138398/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3768/138398/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-09-22_09.50.17/lib.mongooseim.logs/run.2022-09-22_09.50.17/suite.log.html)

File: src/mongoose_admin_api/mongoose_admin_api_domain.erl [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+         content_types_provided/2,
+         content_types_accepted/2,
+         allowed_methods/2,
+         to_json/2,
+         from_json/2,
+         delete_resource/2]).
+
+-ignore_xref([to_json/2, from_json/2]).
+
+-import(mongoose_admin_api, [parse_body/1, try_handle_request/3, throw_error/2, resource_created/4]).
```
Comment (JanuszJakubiec) [inline src/mongoose_admin_api/mongoose_admin_api_domain.erl:15]: This line has 101 characters

File: src/mongoose_admin_api/mongoose_admin_api_domain.erl [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+content_types_provided(Req, State) ->
+    {[
+      {{<<"application">>, <<"json">>, '*'}, to_json}
+     ], Req, State}.
+
+-spec content_types_accepted(req(), state()) ->
+          {[{{binary(), binary(), '*'}, atom()}], req(), state()}.
+content_types_accepted(Req, State) ->
+    {[
+      {{<<"application">>, <<"json">>, '*'}, from_json}
```
Comment (JanuszJakubiec) [inline src/mongoose_admin_api/mongoose_admin_api_domain.erl:39]: Wouldn't it be better to write it in one line?

File: src/mongoose_admin_api/mongoose_admin_api_domain.erl [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+-spec is_authorized(req(), state()) -> {true | {false, iodata()}, req(), state()}.
+is_authorized(Req, State) ->
+    mongoose_admin_api:is_authorized(Req, State).
+
+-spec content_types_provided(req(), state()) ->
+          {[{{binary(), binary(), '*'}, atom()}], req(), state()}.
+content_types_provided(Req, State) ->
+    {[
+      {{<<"application">>, <<"json">>, '*'}, to_json}
```
Comment (JanuszJakubiec) [inline src/mongoose_admin_api/mongoose_admin_api_domain.erl:32]: Wouldn't it be better to write it in one line?

**Review by JanuszJakubiec — APPROVED** [review]
Looks good, I've added some minor suggestions.

File: src/mongoose_admin_api/mongoose_admin_api_domain.erl (hunk 2 above)
Comment (chrzaszcz) [reply src/mongoose_admin_api/mongoose_admin_api_domain.erl:39]: It is like this in all API modules, and I think it is more readable this way (multiple nested tuples).

Comment (JanuszJakubiec) [reply src/mongoose_admin_api/mongoose_admin_api_domain.erl:39]: ok

**Review by arcusfelis — APPROVED** [review]
ok

**Pull request merged**